### PR TITLE
Use setup-fips on SL Micro 6.0

### PR DESCRIPTION
**Pull request by paolostivanin**
Ticket: https://progress.opensuse.org/issues/158499
VRs: [SLEM](https://openqa.suse.de/tests/14155121#step/fips_setup/1), [15-SP6-ker](https://openqa.suse.de/tests/14155120#step/fips_setup/1), [15-SP6-env](https://openqa.suse.de/tests/14155550#step/fips_setup/1), [JeOS](https://openqa.suse.de/tests/overview?distri=sle&version=15-SP5&build=20240425-1&groupid=431)

### CODE REVIEW

File: tests/fips/fips_setup.pm [hunk 1]
```diff
@@ -39,8 +39,11 @@ sub enable_fips {
         validate_script_output("fips-mode-setup --check", sub { m/FIPS mode is enabled\.\n.*\nThe current crypto policy \(FIPS\) is based on the FIPS policy\./ });
     } else {
         if (is_transactional) {
-            change_grub_config('=\"[^\"]*', '& fips=1 ', 'GRUB_CMDLINE_LINUX_DEFAULT');
-            trup_call('--continue grub.cfg');
+            # on SL Micro 6.0+ we only need to reboot, no need to manually change grub.
+            if (is_sle_micro('<6.0')) {
+                change_grub_config('=\"[^\"]*', '& fips=1 ', 'GRUB_CMDLINE_LINUX_DEFAULT');
+                trup_call('--continue grub.cfg');
+            }
```
Comment (realcharmer) [inline tests/fips/fips_setup.pm:46]: Does this have to be a nested condition? What about something like this:

```
if (is_transactional && is_sle_micro('<6.0')) {
}
```

Comment (paolostivanin) [reply tests/fips/fips_setup.pm:46]: actually, I can even remove the `is_transactional`

Comment (paolostivanin) [reply tests/fips/fips_setup.pm:46]: not really, because then with sle micro 6.0 it would execute the else branch which would be wrong

Comment (paolostivanin) [reply tests/fips/fips_setup.pm:46]: but I can simplify it, gimme a sec

**Comment by realcharmer** [comment]
LGTM

**Review by tjyrinki — COMMENTED** [review]
Logic seems correct as far as I can twist my head to it.

**Pull request merged**